src/main/java/xbot/common/controls/actuators/XCANSparkMax.java

```diff
@@ -38,23 +39,27 @@
     final DoubleProperty voltageProp;
     final DoubleProperty currentProp;
 
+    protected final String policeTicket;
+
     protected boolean firstPeriodicCall = true;
 
     public XCANSparkMax(
-        int deviceId, 
+        DeviceInfo deviceInfo, 
         String owningSystemPrefix, 
         String name, 
         PropertyFactory pf, 
         DevicePolice police,
         CommonLibFactory clf,
         XCANSparkMaxPIDProperties defaultPIDProperties) {
         this.clf = clf;
-        this.deviceId = deviceId;
+        this.deviceId = deviceInfo.channel;
         this.propertyFactory = pf;
         this.propertyFactory.setPrefix(owningSystemPrefix);
         this.propertyFactory.appendPrefix(name);
         prefix = pf.getPrefix();
-        police.registerDevice(DeviceType.CAN, deviceId, this);
+        policeTicket = police.registerDevice(DeviceType.CAN, deviceId, this);
+
+        setInverted(deviceInfo.inverted);
```

Comment:
This has to be done in some kind of delayed fashion - if we do this in the constructor, it will attempt to call `setInverted` on CANSparkMaxWpiAdapter, which will attempt to call `internalSpark.setInverted(isInverted)`... but the internalSpark won't have been created yet!

We didn't have a great solution for this before - you might notice that XCANTalon has to deal with the same issue, and it "solves" it by adding some extra configuration methods like `configureMotorTeam` and `configureAsMasterMotor` that are meant to be called after the object is created. There may be a better way to do this, though. Maybe the abstract class sets some flag that the subclass reads during its own constructor?

Reply:
CANTalonWpiAdapter just sets inverted after assigning the internal object. I updated this to follow the same pattern.